### PR TITLE
Add podif-crd in controller config

### PR DESCRIPTION
**Pull request by tanyatukade**

### CODE REVIEW

File: provision/acc_provision/templates/aci-containers.yaml [hunk 1]
```diff
@@ -299,6 +299,19 @@ spec:
     served: true
     storage: true
 ---
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (tomflynn) [inline provision/acc_provision/templates/aci-containers.yaml:302]: Please rework the if checks above to avoid repeating the CRD definition

File: provision/acc_provision/templates/aci-containers.yaml [hunk 2]
```diff
@@ -744,6 +757,12 @@ rules:
   - watch
   - get
   - update
+- apiGroups:
+  - "aci.aw"
+  resources:
+  - podifs
+  verbs:
+  - '*'
```
Comment (tomflynn) [inline provision/acc_provision/templates/aci-containers.yaml:765]: Same comment here. There's set of podifs permissions for 'use_aci_anywhere_crd' that should should be expanded to all cases

**Review by tomflynn — CHANGES_REQUESTED** [review]

File: provision/acc_provision/templates/aci-containers.yaml (hunk 1 above)
Comment (tanyatukade) [reply provision/acc_provision/templates/aci-containers.yaml]: removed the check for podif to resuse the crd in non-overlay case as well.

File: provision/acc_provision/templates/aci-containers.yaml (hunk 2 above)
Comment (tanyatukade) [reply provision/acc_provision/templates/aci-containers.yaml:751]: done

Comment (tanyatukade) [reply provision/acc_provision/templates/aci-containers.yaml:751]: in the aci_anywhere check, the permissions are '*'. Should that be expanded?

File: provision/testdata/base_case_tar/cluster-network-22-ClusterRole-aci-containers:host-agent.yaml [hunk 3]
```diff
@@ -86,12 +93,6 @@ rules:
   - list
   - watch
   - get
-- apiGroups:
```
Comment (tomflynn) [inline provision/testdata/base_case_tar/cluster-network-22-ClusterRole-aci-containers:host-agent.yaml:89]: This change doesn't seem related...merge issue?

**Review by tomflynn — COMMENTED** [review]
Looks good but please take a look at that one unexpected yaml change

File: provision/testdata/base_case_tar/cluster-network-22-ClusterRole-aci-containers:host-agent.yaml (hunk 3 above)
Comment (tomflynn) [reply provision/testdata/base_case_tar/cluster-network-22-ClusterRole-aci-containers:host-agent.yaml:89]: Actually this might just be git incorrectly detecting a rename...

**Comment by snaiksat** [comment]
There are some files getting renamed (more specifically the file order number changing) in the cloud tar. I am wondering why? This would typically happen if new resources are getting added which would result in new manifest files those resources.

**Comment by tanyatukade** [comment]
I manually checked for the config.openshift.io config and it is not deleted. It exists in the cluster-network-22 in openshift_45_esx folder.

**Comment by tanyatukade** [comment]
Will open a new PR since resolving conflicts manually for all the testdata files is time consuming.
